### DISCUSSION TITLE
Nodered within Homeassistant or in docker ? pro's and con's

### DISCUSSION THREAD

**Discussion by Jens-Wymeersch** [zachowj/node-red-contrib-home-assistant-websocket#1670] [General]
I've been experiencing a lot of problems these last few weeks with NR within homeassistant. I start to think, it might be better to have a clean install on docker and connect it to homeassistant.

It would be great to hear, the folks who moved from one to the other. And their experiences.
I assume, you can't use the NR companion anymore ?

**Comment by zachowj** [comment]
You can use the Node-RED Companion with any Home Assistant installation type, as it's simply a custom component for Home Assistant.

**Reply by Jens-Wymeersch** [reply]
So if I run a NR instance in a docker and connect it to home assistant, the companion integration will still work 
Can you run multiple NR instances simultaneously and connect to one homeassistant instance ?
What are the disadvantages of running a NR outside of homeassistant ? Which one do you prefer and why ?

**Reply by zachowj** [reply]
> So if I run a NR instance in a docker and connect it to home assistant, the companion integration will still work

yes

> Can you run multiple NR instances simultaneously and connect to one homeassistant instance ?

yes

> What are the disadvantages of running a NR outside of homeassistant ?

I don't think there is a single disadvantage to running NR in docker vs as an add-on. I guess one would be that you can't access the NR UI via the ingress proxy from the HA UI.

> Which one do you prefer and why ?

I have never ran NR as an add-on for daily use.

**Reply by Jens-Wymeersch** [reply]
Would you suggest running Node-red in a docker or a LXC under proxmox ? And why ?
